### PR TITLE
Add router

### PR DESCRIPTION
**Pull request by goohooh**
1. Applying react-router to App
2. Following the hooks rule? Let me know your thoughts. ([React docs](https://reactjs.org/docs/hooks-rules.html)

### CODE REVIEW

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -1,30 +1,33 @@
...
-          <IndexPage
-            search={BookLister}
-            statusFetcher={getResourceStatus}
-            currentUser={currentUser}/> :
-          <SignInPage />
-      }
+        <CssBaseline />
+        {loadingCurrentUser ?
+          <div>Loading...</div> :
+            <Router>
```
Comment (Goblinlordx) [inline frontend/src/App.tsx]: Looks like you can do the ternary above the router.

File: frontend/src/App.tsx [hunk 2]
```diff
@@ -1,30 +1,33 @@
...
+              <Switch>
+                <Route path='/' component={() => {
```
Comment (Goblinlordx) [inline frontend/src/App.tsx]: It looks like you should use `render` here instead of `component` prop.

**Comment by Goblinlordx** [comment]
tests~

**Comment by goohooh** [comment]
@Goblinlordx I Added memory router to each component. But I'm not sure it is best solution

File: frontend/src/components/SignInPage/SignInPage.test.tsx [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
+import { MemoryRouter } from 'react-router-dom';
 
 import SignInPage from "./SignInPage";
 
 it("renders at least one button", () => {
-  const instance = renderer.create(<SignInPage />).root;
+  const instance = renderer.create(
+    <MemoryRouter>
+      <SignInPage />
+    </MemoryRouter>
```
Comment (Goblinlordx) [inline frontend/src/components/SignInPage/SignInPage.test.tsx]: This definitely seems unnecessary.  SignInPage is not rendered inside a Router, why is this needed?

File: frontend/src/components/IndexPage/IndexPage.tsx [hunk 4]
```diff
@@ -86,8 +87,8 @@ const IndexPage: React.SFC<IndexPageProps> = ({
           </SearchResult>
         </SearchForm>
       </SearchContainer>
-    </div>
+    </>
   );
 };
 
-export default IndexPage;
+export default withRouter(IndexPage);
```
Comment (Goblinlordx) [inline frontend/src/components/IndexPage/IndexPage.tsx]: This does not seem necessary.  If route props are needed, they can be passed in render where the routing happens.

**Comment by Goblinlordx** [comment]
I think you should make sure you are using `withRouter` appropriately and only where necessary.  It looks like it is wrapping components that don't need and should not have it.

**Comment by goohooh** [comment]
@Goblinlordx I misunderstood about `RouteComponentProps`

**Comment by Goblinlordx** [comment]
Much better ^^ :+1:
